src/utils/blogs.ts

```diff
@@ -92,21 +99,33 @@ export async function getBlogBySlug<F extends (keyof Blog)[]>(
   const date = typeof data.date === 'string' ? data.date : fs.statSync(fullPath).birthtime.toISOString()
   const readingTime = Math.round(content.length / 1300).toString()
 
-  const authorNames: string[] = Array.isArray(data.author)
-    ? data.author.filter((author): author is string => typeof author === 'string')
-    : typeof data.author === 'string'
-    ? [data.author]
-    : ['Nervos']
-
-  const authors = authorNames.map(name => {
-    if (name.toLowerCase() === 'nervos') return { name, avatar: '/images/nervos_avatar.svg' }
-    if (name.startsWith('github:')) {
-      name = name.substring('github:'.length)
-      return { name, avatar: `https://avatars.githubusercontent.com/${name}` }
-    }
+  const list = Array.isArray(data.author) ? data.author : [data.author]
+  const authors: { name: string; avatar?: string }[] = list
+    .filter(item => !!item)
+    .map(author => {
+      if (typeof author === 'string') {
+        // string authors
+        if (author.startsWith('github:')) {
+          // github author
+          const name = author.substring('github:'.length)
+          return { name, avatar: `https://avatars.githubusercontent.com/${name}` }
+        } else {
+          // default string author
+          return { name: author }
+        }
+      } else if (Object.prototype.toString.call(author) === '[object Object]') {
+        // object authors
+        const name = (author as AuthorDetail).name
+        return { name, avatar: `https://avatars.githubusercontent.com/${name}` }
+      } else {
+        return { name: author as string }
+      }
```

Comment:
These 2 branches can be simplified as

```js
const name = author?.name
const avatar = name ? `https://avatars.githubusercontent.com/${name}` : null
return {
  name,
  avatar
}
```

Reply:
And the 3rd branch is only triggered when `author` is `undefined`, which should is filtered out in the `filter` step